### PR TITLE
config/ingress: add ingress publishing scope config

### PR DESCRIPTION
**Pull request by ironcladlou**
Sometimes users need to ensure their ingresscontrollers are always internal on
cloud platforms, even during installation (e.g. security policy, cost, resource
quotas).

This commit introduces a way for installer users to express the intent that
ingresscontrollers should default to internal instead of public.

### CODE REVIEW

**Comment by ironcladlou** [comment]
Compare to:
* https://github.com/openshift/enhancements/pull/31
* https://github.com/openshift/enhancements/pull/25

**Comment by ironcladlou** [comment]
@deads2k @abhinavdahiya @openshift/sig-network-edge

File: config/v1/types_ingress.go [hunk 1]
```diff
@@ -32,8 +32,31 @@ type IngressSpec struct {
...
+
+	// scope is the default publishing scope for new ingress controllers on this
+	// cluster which are exposed by an OpenShift-managed load balancer.
+	//
+	// The default is "External".
+	//
+	// Changes to this field after creation will only be recognized by new
+	// ingress controllers.
+	// +optional
+	Scope *LoadBalancerScope `json:"scope,omitempty"`
```
Comment (ironcladlou) [inline config/v1/types_ingress.go]: Is there any escaping using a pointer in this case?

**Comment by ironcladlou** [comment]
@smarterclayton maybe this is where we have to meet in the middle? https://github.com/openshift/enhancements/pull/25 and https://github.com/openshift/enhancements/pull/31 seem unexpectedly controversial in their own ways.

File: config/v1/types_ingress.go (hunk 1 above)
Comment (deads2k) [reply config/v1/types_ingress.go]: > Is there any escaping using a pointer in this case?

This would be no pointer, allow an empty string.

**Comment by deads2k** [comment]
/hold

this is still under discussion.  I prefer keeping the values low level.

**Comment by deads2k** [comment]
We aren't going to add a top level config option.

/close

**Pull request closed without merging**